packages/block-editor/src/components/block-list/block.js

```diff
@@ -159,7 +160,8 @@ function BlockListBlock( {
 			'is-focused':
 				isFocusMode && ( isSelected || isAncestorOfSelectedBlock ),
 			'is-focus-mode': isFocusMode,
-			'has-child-selected': isAncestorOfSelectedBlock,
+			// Don't select child while dragging.
+			'has-child-selected': isAncestorOfSelectedBlock && ! isDragging,
```

Comment:
Not sure the comments here add much, I think we could either remove them or add some more detail on why those classes aren't applied when dragging.